primitives/src/zeitgeist_multi_reservable_currency.rs

```diff
@@ -21,24 +21,16 @@ where
 {
     fn accounts_by_currency_id(
         currency_id: Self::CurrencyId,
-    ) -> (u32, Vec<(T::AccountId, AccountData<T::Balance>)>) {
-        let mut total: u32 = 0;
-        (
-            total,
-            <Accounts<T>>::iter()
-                .filter_map(|(k0, k1, v)| {
-                    if total < u32::MAX {
-                        total += 1;
-                    }
-
-                    if k1 == currency_id {
-                        Some((k0, v))
-                    } else {
-                        None
-                    }
-                })
-                .collect(),
-        )
```

Comment:
IINM, `(u32, Vec<(T::AccountId, AccountData<T::Balance>)>)` here is the same of
```
let rslt = accounts_by_currency_id(something);
(rslt.len(), rslt)
```

Reply:
This is not the same, since the `u32` that was returned before was the number of total accounts and `rslt.len()` is the number of accounts that hold a specific `CurrencyId`. Please revert.

Reply:
OK, in any case `total` was and always will be zero so is this parameter really necessary?
If so, I will fix this snippet to return the total number of accounts.

Reply:
@c410-f3r What a bummer! In that case can you restructure the code that it takes the actual total number after `<Accounts<T>>::iter().filter_map` was evaluated?
```
        let mut total: u32 = 0;
        let accounts = <Accounts<T>>::iter()
        .filter_map(|(k0, k1, v)| {
            if total < u32::MAX {
                total += 1;
            }

            if k1 == currency_id {
                Some((k0, v))
            } else {
                None
            }
        })
        .collect();
        (total, accounts)
```

Reply:
I don't like this whole construct but we need to count the number of total accounts currently to properly calculate the weight. Unfortunately getting the number of total accounts does require iterating over every account (to my knowledge), so I have decided to return this information from this function to avoid one additional iteration over every account.

Reply:
Oh, don't worry. I was concerned about possible unintentional leftovers but this snippet is indeed used and expected in other places.
Thanks for the review 👍. Tomorrow everything will probably be fixed.